### DISCUSSION TITLE
Accesso Remoto da App con Dyndns

### DISCUSSION THREAD

**Discussion by Borghese** [indomus/forum#2681] [HUB: Home Assistant]
ciao a tutti,
grazie alle vostre guide ho installato HA OS su un raspberry PI4 e adesso inizio a smanettare con tutta la sensoristica che ho in casa per domotizzarla!
Ho installato anche gli add on "imprescindibili" ma mi sono fermato un attimo sul tema di accesso remoto : 
Leggevo che ormai duckdns non è più sicuro ed è superato
Vorrei evitare la vpn, perchè sennò da smartphone, ogni volta dovrei prima attivarla, prima di avviare l'app di ha
Volevo capire insieme a voi se ci fosse la possibilità di configurare un accesso remoto, utilizzando un dominio che già possiedo (con dyndns appunto, e che pago annualmente) poi forwardando sulla porta 8123
sapete dove posso trovare qualche guida che mi aiuti in questa configurazione ?
grazie in anticipo

**Comment by snakuzzo** [comment]
Ciao.
L'utilizzo di duckdns non è né insicuro né superato.
Diciamo che ci sono metodi migliori e più sicuri per esporre su internet il proprio servizio (in questo caso Home Assistant immagino).
Detto ciò, potresti utilizzare sicuramente il tuo attuale dominio, ma dovresti:

- assicurarti che il RECORD A sia sempre aggiornato con il tuo "attuale" IP pubblico: risultato facilmente ottenibile utilizzando le API di DynDNS (dovresti trovare tutto nelle FAQ sul sito) e/o utilizzare la funzionalità di DDNS solitamente presente in tutti i modem/router (da verificare se è supportato DynDNS).

- avere un certificato rilasciato da una CA valida e riconosciuta da utilizzare: in tal caso il consiglio sarebbe quello di utilizzare un reverse proxy come Caddy ([QUI](https://indomus.it/guide/installare-e-configurare-caddy-su-home-assistant-os-hassos-hassio/) la guida) o NGINX Proxy Manager (basta che cerchi su Google) che ti permettono, dopo il primo setip, di "dimenticarti" di questa cosa del certificato, visto che sarà automaticamente rilasciato aggiornato ogni 3 mesi prima della scadenza 

- Configurare il port forwarding per mappare la porta TCP esterna (es. 443) verso la TCP interna 8123 del tuo server (ovviamente dovrai assicurarti che il tuo server abbia sempre lo stesso IP LAN)

Soluzione alternativa e più sicura invece... utilizzare il tunneling di Cloudflare che ti permette di non fare nessuno dei passaggi successivi, oltre ad offrirti tutta una serie di vantaggi a livello di sicurezza:

Guida step 1 -> [QUI](https://indomus.it/guide/configurare-cloudflare-per-il-proprio-dominio-personale-per-protezione-reverse-proxy-cdn-e-altro/)
Guida step 2 -> [QUI](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-tradizionali-o-che-usino-nat-via-cloudflare/)

Inutile dire che la soluzione migliore è quella relativa a Cloudflare

**Comment by Borghese** [comment]
grazie snakuzzo! 
Quindi mi sembra di capire sia molto più consigliato cloudflare. 
Ok mi do una lettura alle guide e procedo in tal senso allora
grazie ancora!
